src/Services/MiningRewardsService.ts

```diff
@@ -0,0 +1,21 @@
+import {MiningReward} from "../Models/MiningReward";
+import config from "../Config/Config";
+import {Node} from "../Models/Node";
+import {MiningRewardInput} from "../Types/MiningRewardType";
+
+export class MiningRewardsService {
+
+    public async storeMiningRewards(nodes: Array<Node>, rewards: Array<MiningRewardInput>) {
+        return await Promise.all(nodes.map(
+            (node: Node, index) => (
+                {
+                    cid: rewards[index].cid,
+                    rewardAmount: config.rewardAmount,
+                    nodeId: node.id,
+                }
+            )
+        ).map(async (node) => {
```

Comment:
You don't need the second mapping and interate twice, you can do the create immediatelly.

```
(node: Node, index) => {
  const nodeToSave = {
                   cid: rewards[index].cid,
                    rewardAmount: config.rewardAmount,
                    nodeId: node.id,
  }
return MiningReward.create(nodeToSave);
}
```